packages/internal-workflow-common/src/activity-options.ts

```diff
@@ -87,6 +80,80 @@ Note that the Temporal Server doesn't detect Worker process failures directly. I
+   */
+  activityId?: string;
+
+  /**
+   * RetryPolicy that define how activity is retried in case of failure. If this is not set, then the server-defined default activity retry policy will be used. To ensure zero retries, set maximum attempts to 1.
+   */
+  retry?: RetryPolicy;
+
+  /**
+   * Maximum time the local activity is allowed to execute after the task is dispatched. This
+   * timeout is always retryable.
+   *
+   * Either this option or {@link scheduleToCloseTimeout} is required.
+   * If set, this must be <= {@link scheduleToCloseTimeout}, otherwise, it will be clamped down.
+   *
+   * @format {@link https://www.npmjs.com/package/ms | ms} formatted string or number of milliseconds
+   */
+  startToCloseTimeout?: string | number;
+
+  /**
+   * Limits time the local activity can idle internally before being executed. That can happen if
+   * the worker is currently at max concurrent local activity executions. This timeout is always
+   * non retryable as all a retry would achieve is to put it back into the same queue. Defaults
+   * to {@link scheduleToCloseTimeout} if not specified and that is set. Must be <=
+   * {@link scheduleToCloseTimeout} when set, otherwise, it will be clamped down.
+   *
+   * @default unlimited
+   * @format {@link https://www.npmjs.com/package/ms | ms} formatted string or number of milliseconds
+   */
+  scheduleToStartTimeout?: string | number;
+
+  /**
+   * Indicates how long the caller is willing to wait for local activity completion. Limits how
+   * long retries will be attempted. When not specified defaults to the workflow execution
+   * timeout (which may be unset).
+   *
+   * Either this option or {@link startToCloseTimeout} is required.
+   *
+   * @default unlimited
+   * @format {@link https://www.npmjs.com/package/ms | ms} formatted string or number of milliseconds
+   */
+  scheduleToCloseTimeout?: string | number;
+
+  /**
+   * If the activity is retrying and backoff would exceed this value, lang will be told to
+   * schedule a timer and retry the activity after. Otherwise, backoff will happen internally in
+   * core.
+   *
+   * @default 1 minute
+   * @format {@link https://www.npmjs.com/package/ms | ms} formatted string or number of milliseconds
+   **/
+  localRetryThreshold?: string | number;
+
+  /**
+   * Determines what the SDK does when the Activity is cancelled.
+   * - `TRY_CANCEL` - Initiate a cancellation request and immediately report cancellation to the workflow.
+   * - `WAIT_CANCELLATION_COMPLETED` - Wait for activity cancellation completion. Note that activity must heartbeat to receive a
+   *   cancellation notification. This can block the cancellation for a long time if activity doesn't
+   *   heartbeat or chooses to ignore the cancellation request.
+   * - `ABANDON` - Do not request cancellation of the activity and immediately report cancellation to the workflow.
+   */
+  cancellationType?: coresdk.workflow_commands.ActivityCancellationType;
```

Comment:
This doesn't exist for local activities in Java and Go (but if it has value in TS, cool)

Reply:
Same reasoning as the other thing, but, arguably this is actually meaningfully useful. I think it's worth exposing.

Reply:
+1 to exposing in other SDKs